packages/codebytes/src/editor.tsx

```diff
@@ -38,16 +39,17 @@ type EditorProps = {
   hideCopyButton: boolean;
   language: languageOption;
   text: string;
+
   onChange: (text: string) => void;
-  onCopy?: (text: string, language: string) => void;
+  on?: Pick<CodebytesChangeHandlerMap, 'copy' | 'run'>;
```

Comment:
are we doing this pick because we haven't implemented the other handlers yet? if so, i'd veer on the side of not adding the other handlers to the type until they're implemented. 
also do we have tickets to implement the other handlers, @BandanaKM?

Reply:
oh wait looks we do already use them all, so why do we need the pick here?

Reply:
@borisonr, interesting question. in this pr, all the handlers are implemented, so we're pretty complete. we are using pick here to select the types for the `copy` and `run` functions, which are the only handlers being passed in as props and used in the `Editor` component directly. 

if we really didn't like `Pick` we could pass onCopy and onRun in directly. we could really go either way. 
```
<Editor
     onCopy={on?.copy}
     onRun={on?.run}
/>
```

Reply:
it's not that i don't like pick, i just dont think it's necessary..? we're passing in the whole object right?

Reply:
Actually, you are right, we may not need it in this case! Good catch. It'll make the type very very specific for the editor component to use pick, but it's not necessary at all.